packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -247,6 +301,64 @@ class ImageCache {
   }
 }
 
+/// A class that provides information about how the [ImageCache] is
+/// tracking an image.
+///
+/// A [pending] image is one that has not completed yet. It may also be tracked
+/// as [live] because something is listening to it, but it has not yet
+/// [completed].
+///
+/// A [completed] image is being held in the LRU cache, and is subject to eviction
+/// based on [ImageCache.maximumSizeBytes] and [ImageCache.maximumSize]. It may
+/// [live], but not [pending].
+///
+/// A [live] image is being held until its [ImageStreamCompleter] has no more
+/// listeners. It may also be [pending] or [completed].
+///
+/// An [untracked] image is not being cached.
+///
+/// To obtain an [ImageCacheLocation], use [ImageCache.locationForKey] or
+/// [ImageProvider.findCacheLocation].
+class ImageCacheLocation {
```

Comment:
Mark this class as `@immutable` since it overrides hashCode and operator==?

Reply:
That will mean that someone who writes their own image cache and re-implements this class can't make it stateful.  I'm not sure we want to enforce that here (as in, I don't see what the harm would be in letting someone make this a stateful class).